core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -123,7 +123,9 @@ class LogManager(logDirs: Seq[File],
   }
 
   private val dirLocks = lockLogDirs(liveLogDirs)
-  val directoryIds: Map[String, Uuid] = loadDirectoryIds(liveLogDirs)
+  private val directoryIds: mutable.Map[String, Uuid] = loadDirectoryIds(liveLogDirs)
+  def directoryIdsSet: Predef.Set[Uuid] = directoryIds.values.toSet
+
   @volatile private var recoveryPointCheckpoints = liveLogDirs.map(dir =>
```

Comment:
You need to rebase your changes on top of latest apache/trunk
This change needs to be applied to 'core/src/main/scala/kafka/server/KafkaServer.scala' which was updated pass this set to `BrokerLifecycleManager` if the Zk broker has migration enabled.

Reply:
rebased and fixed few tests